### PR TITLE
Improve Advisor API

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Add an `advise` function that is easier to use programmatically, see the commit messages for details.

**Breaking change:**
The function `Advisor.retrieveFindings()` was renamed to `Advisor.advise()`.

### CODE REVIEW

File: advisor/src/main/kotlin/Advisor.kt [hunk 1]
```diff
@@ -51,7 +51,7 @@ class Advisor(
     }
 
     @JvmOverloads
-    fun retrieveFindings(ortResult: OrtResult, skipExcluded: Boolean = false): OrtResult {
+    fun advise(ortResult: OrtResult, skipExcluded: Boolean = false): OrtResult {
```
Comment (sschuberth) [inline advisor/src/main/kotlin/Advisor.kt:59]: Commit message: "This align*s*..."

**Review by sschuberth — CHANGES_REQUESTED** [review]

File: advisor/src/main/kotlin/Advisor.kt (hunk 1 above)
Comment (oheger-bosch) [reply advisor/src/main/kotlin/Advisor.kt:59]: While you are at it, would it make sense to make the `advise` function also a suspend function analogously to the API of the scanner? The advise providers already are suspendable.

Comment (mnonnenmacher) [reply advisor/src/main/kotlin/Advisor.kt:59]: @sschuberth  Fixed, I also added some commitlint fixes.

**Review by sschuberth — APPROVED** [review]

**Review by oheger-bosch — APPROVED** [review]
The changes at the advisor are fine with me. Regarding the commit lint fixes, I leave the review to you guys with more experience in this area.

File: advisor/src/main/kotlin/Advisor.kt (hunk 1 above)
Comment (oheger-bosch) [reply advisor/src/main/kotlin/Advisor.kt:59]: @mnonnenmacher, if you want, I could create a corresponding PR.

Comment (sschuberth) [reply advisor/src/main/kotlin/Advisor.kt:59]: I'd appreciate to see that, too, @oheger-bosch 😁

**Pull request merged**